### PR TITLE
feat: Add support for pip Package Manager

### PR DESCRIPTION
**Pull request by Sahilb315**

### CODE REVIEW

File: packagemanager/npm.go [hunk 1]
```diff
@@ -37,6 +37,8 @@ func NewNpmPackageManager(config NpmPackageManagerConfig) (*npmPackageManager, e
 	}, nil
 }
 
+var _ PackageManager = &npmPackageManager{}
```
Comment (abhisek) [inline packagemanager/npm.go:40]: Nice. Thanks!

**Comment by abhisek** [comment]
Thanks for picking this up @Sahilb315 Let me know once ready for review

File: packagemanager/pypi_resolver.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	if err != nil {
+		return "", fmt.Errorf("no version matches constraint %q: %w", versionConstraint, err)
+	}
+
+	return bestMatch.Original(), nil
+}
+
+// pipFetchPackageVersionsInfo retrieves package information from PyPI
+func pipFetchPackageVersionsInfo(packageName string) (*PyPIPackage, error) {
+	url := fmt.Sprintf("https://pypi.org/pypi/%s/json", packageName)
```
Comment (abhisek) [inline packagemanager/pypi_resolver.go]: We are already doing this from the `packageregistry.Client` isn't it?

Comment (Sahilb315) [reply packagemanager/pypi_resolver.go]: Yes, replaced it with our packageregistry.

**Comment by Sahilb315** [comment]
> @Sahilb315 If its ready for review, please mark it so. Currently I see its marked as draft

There are still some edge cases left to test out. After testing will mark as ready for review

**Comment by Sahilb315** [comment]
@abhisek Please review when you get a chance. Will add test cases if any functionality needs coverage.

File: cmd/npm/npm.go [hunk 3]
```diff
@@ -32,6 +33,25 @@ func executeNpmFlow(ctx context.Context, args []string) error {
 	if err != nil {
 		ui.Fatalf("Failed to create npm package manager proxy: %s", err)
 	}
+	config, err := config.FromContext(ctx)
```
Comment (OmkarPh) [inline cmd/npm/npm.go:37]:
```suggestion

	config, err := config.FromContext(ctx)
```

File: packagemanager/npm_resolver.go [hunk 4]
```diff
@@ -78,7 +78,9 @@ func (r *npmDependencyResolver) ResolveDependencies(ctx context.Context,
 		TransitiveDepth:               r.config.TransitiveDepth,
 		FailFast:                      r.config.FailFast,
 		MaxConcurrency:                r.config.MaxConcurrency,
-	}, npmCleanVersion)
+	}, func(packageName, version string) string {
```
Comment (OmkarPh) [inline packagemanager/npm_resolver.go]: i think its a good idea to put the function in a separate variable or declare it separately
and use that for better readability

File: packagemanager/pypi.go [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+		return version
+	}
+
+	version = strings.TrimPrefix(version, "~=")
+	parts := strings.Split(version, ".")
+	if len(parts) < 2 {
+		return "" // invalid
+	}
+
+	if len(parts) == 2 {
```
Comment (OmkarPh) [inline packagemanager/pypi.go]: switch case for this ?

File: pkg/utils/utils.go [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package utils
+
+func Contains(slice []string, item string) bool {
```
Comment (OmkarPh) [inline pkg/utils/utils.go]: use `slices.Contains`

File: cmd/npm/npm.go (hunk 3 above)
Comment (Sahilb315) [reply cmd/npm/npm.go:37]: both similar

Comment (OmkarPh) [reply cmd/npm/npm.go:37]: good to have "assignment & err-check" blocks separated by an extra line
just a minor nit for readability

File: cmd/npm/pnpm.go [hunk 7]
```diff
@@ -32,6 +34,22 @@ func executePnpmFlow(ctx context.Context, args []string) error {
 	if err != nil {
 		ui.Fatalf("Failed to create pnpm package manager proxy: %s", err)
 	}
+	config, err := config.FromContext(ctx)
```
Comment (OmkarPh) [inline cmd/npm/pnpm.go:38]:
```suggestion

	config, err := config.FromContext(ctx)
```

File: cmd/pypi/pip.go [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+		},
+	}
+}
+
+func executePipFlow(ctx context.Context, args []string) error {
+	packageManager, err := packagemanager.NewPipPackageManager(packagemanager.DefaultPipPackageManagerConfig())
+	if err != nil {
+		return fmt.Errorf("failed to create pip package manager: %w", err)
+	}
+	config, err := config.FromContext(ctx)
```
Comment (OmkarPh) [inline cmd/pypi/pip.go:37]:
```suggestion

	config, err := config.FromContext(ctx)
```

**Review by OmkarPh — APPROVED** [review]
